src/CoreFoundation/Dispatch.cs

```diff
@@ -427,11 +415,8 @@ public void Submit (Action<int> action, long times)
 		// dispatch_queue_t dispatch_get_global_queue (long priority, unsigned long flags);
 		extern static IntPtr dispatch_get_global_queue (nint priority, nuint flags);
 
-#if MONOMAC
-		[Obsoleted (PlatformName.MacOSX, 10, 7)]
 		[DllImport (Constants.libcLibrary)]
 		extern static IntPtr dispatch_get_main_queue ();
```

Comment:
according to web docs it's new in iOS 8 and macOS 10.10
weird that it would have been deprecated in 10.7 ?!?
let's run this into old sim/devices

Reply:
https://developer.apple.com/documentation/dispatch/1452921-dispatch_get_main_queue

Reply:
It seems my grepping was flawed: `dispatch_get_main_queue` shows up in earlier iOS SDKs, but it's a define, not an actual function (which it didn't become until iOS 8).

I'll have a look and see if we can remove the call to `dispatch_get_main_queue` entirely, and just rely on the `_dispatch_main_q` variable instead.

Reply:
Some history:

https://github.com/xamarin/maccore/commit/25097444a9eacab88fe9b0d7f3ae640ce1768d14
https://github.com/xamarin/maccore/commit/04ea3ac8642812a94f401c286e62e67e2f1920dc